feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt

```diff
@@ -86,6 +88,46 @@ internal class AmazonBilling constructor(
         )
     }
 
+    override fun getProperProductID(
+        productID: String,
+        purchaseToken: String,
+        amazonUserID: String?,
+        onSuccess: (properProductID: String) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        if (amazonUserID == null) {
+            onError(missingAmazonUserIdError())
+            return
+        }
+
+        val currentlyCachedTokensToSkus = cache.getReceiptSkus()
+
+        if (currentlyCachedTokensToSkus.containsKey(purchaseToken)) {
```

Comment:
i think you could move this outer if, the ?.let is basically an if?

Reply:
oh yes, I can do:

```
        currentlyCachedTokensToSkus[purchaseToken]?.let { sku ->
            onSuccess(sku)
            return
        }
```